python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -582,17 +587,150 @@ def on_sso_disable_renewal(self, event):
+                # that we are using cookies from a multi-session environment. We
+                # have no way to identify the proper user id in the lot.
+                raise Saml2SssoMultiSessionNotSupportedError('Multi-session not supported')
+            user_id = cookie[28:]
+    return user_id
+
+
+def _get_cookie_from_prefix(encoded_cookies, cookie_prefix):
+    """
+    Returns a cookie value based on a prefix to which we will append the user id.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+    :param cookie_prefix: The prefix of the cookie name.
+
+    :returns: A string of the cookie value, or None.
+    """
+    value = None
+    cookies = _decode_cookies(encoded_cookies)
+    key = "%s%s" % (cookie_prefix, _get_shotgun_user_id(cookies))
+    if key in cookies:
+        value = cookies[key].value
+    return value
+
+
+def is_sso_enabled(encoded_cookies):
+    """
+    Indicate if SSO is being used from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: True or False
+    """
+    cookies = _decode_cookies(encoded_cookies)
+    return _get_shotgun_user_id(cookies) is not None
+
+
+def get_saml_claims_expiration(encoded_cookies):
+    """
+    Obtain the expiration time of the saml claims from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: An int with the time in seconds since January 1st 1970 UTC, or None
+    """
+    saml_claims_expiration = _get_cookie_from_prefix(encoded_cookies, "shotgun_sso_session_expiration_u")
+    if saml_claims_expiration is not None:
+        saml_claims_expiration = int(saml_claims_expiration)
+    return saml_claims_expiration
+
+
+def get_saml_user_name(encoded_cookies):
+    """
+    Obtain the saml user name from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: A string with the user name, or None
+    """
+    user_name = _get_cookie_from_prefix(encoded_cookies, "shotgun_sso_session_userid_u")
+    if user_name is not None:
+        user_name = urllib.unquote(user_name)
+    return user_name
+
+
+def get_session_id(encoded_cookies):
+    """
+    Obtain the session id from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: A string with the session id, or None
+    """
+    session_id = None
+    cookies = _decode_cookies(encoded_cookies)
+    key = "_session_id"
+    if key in cookies:
+        session_id = cookies[key].value
+    return session_id
+
+
+def get_csrf_token(encoded_cookies):
+    """
+    Obtain the csrf token from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: A string with the csrf token, or None
+    """
+    return _get_cookie_from_prefix(encoded_cookies, "csrf_token_u")
+
+
+def get_csrf_key(encoded_cookies):
```

Comment:
i think adding a class and turning all these getters into properties will make this cleaner!